[Issue 402] Updates README.md and Supporting docs (#433)
HHS/grants-equity: README.md

# Grants Equity

A modernization effort for Grants.gov.

## About the Project

Our vision is for the following to become true:

> Grants.gov is the simplest, most inclusive, and most gratifying way to find and apply for financial assistance ever built, inside or outside the federal government.

Reviewer: ```suggestion
_**Grants.gov is the simplest, most inclusive, and most gratifying way to find and apply for financial assistance ever built, inside or outside the federal government.**_
```

Grants.gov helps ensure that no communities are underserved by the federal government.

See [goals.md](./documentation/goals.md) for more information about the vision and goals for the project.

## Core Team

> The core team on the project is a small group of designers, developers, product managers, and more  working for an with the Department of Health and Human Services and other federal agencies.

Reviewer: ```suggestion
The core team on the grants.gov project is a small group of designers, developers, and product managers working for and with the Department of Health and Human Services, and other federal agencies, and community volunteers.
```

Reviewer: I updated the language from the runbook

> An up-to-date list of core team members can be found in [MAINTAINERS.md](./MAINTAINERS.md). At this time, the project is still building the core team and defining roles and responsibilities. We are eagerly seeking individuals who would like to join the community and help us define and fill these roles. 

Reviewer 2: We might need to update ./maintainers.md, do we want to do that in this PR or a separate one?

Author: We just updated that file: https://github.com/HHS/grants-equity/blob/main/MAINTAINERS.md#maintainers :)

## Repository Structure

> - [./github](./github) contains Github specific settings files and testing, linting, and CI/CD workflows

Reviewer 2: ```suggestion
- [./.github](./.github) contains Github specific settings files and testing, linting, and CI/CD workflows
```

Reviewer 2: Link doesn't work without `.`
